src/model/string-pool.cpp

```
@@ -52,120 +54,121 @@ FieldId Fields::emplace(std::string_view const& str)
     }
     {
         std::unique_lock stringStoreWriteAccess_(stringStoreMutex_);
-        auto [it, insertionTookPlace] = idForString_.emplace(lowerCaseStr, nextId_);
+        auto [it, insertionTookPlace] = idForString_.try_emplace(lowerCaseStr, nextId_);
         if (insertionTookPlace) {
-            stringForId_.emplace(nextId_, str);
-            byteSize_ += (int64_t)str.size();
+            (void)stringForId_.try_emplace(nextId_, str);
+            byteSize_ += static_cast<int64_t>(str.size());
             ++cacheMisses_;
             ++nextId_;
```

Let us add a safety check here: E.g. `if (nextId_ < it->second)` then it wrapped around. In this case, we should throw an exception. We can also statically assert that `StringId` is an unsigned type - this is necessary for the windowed String Pool update streaming to work.